src/components/smart/task.component.js

```diff
@@ -5,42 +5,63 @@ export class Task extends Component {
    *
    * @param title
    * @param description
-   * @param {Function(<Task>)} deleteListItem
+   * @param {Function(<Task>)} deleteTaskListItem
    */
-  constructor(title, description, deleteListItem) {
+  constructor(title, description, deleteTaskListItem, cardRender) {
     super();
     this.componentName = 'task';
     this.title = title;
     this.description = description;
-    this.createTime = new Date();
     this.isCompleted = false;
-    this.deleteListItem = deleteListItem;
+    this.deleteTaskListItem = deleteTaskListItem;
+    this.cardRender = cardRender;
   }
 
   /**
    *
    * @returns {void}
    */
   setCompleted() {
-    this.isCompleted = true;
+    this.isCompleted = !this.isCompleted;
+    this.cardRender();
   }
 
   /**
    *
    * @returns {void}
    */
   deleteTask() {
-    this.deleteListItem(this);
+    this.deleteTaskListItem(this);
+    this.elemRef.remove();
```

Comment:
this dom node should be removed after parent render, so it is not necessary to remove one more time